src/java.base/share/classes/java/math/MutableBigInteger.java

```diff
@@ -699,14 +707,29 @@ private int mulsubBorrow(int[] q, int[] a, int x, int len, int offset) {
      * Assumes that intLen > 0, n > 0 for speed
      */
     private final void primitiveRightShift(int n) {
+        primitiveRightShift(n, value, offset);
+    }
+
+    /**
+     * Right shift this MutableBigInteger n bits, where n is
+     * less than 32, placing the result in the specified array.
+     * Assumes that intLen > 0, n > 0 for speed.
+     * The result can be the value array of this MutableBigInteger,
+     * but for speed the copy is not performed safely, so, in that case
+     * the caller has to make sure that
+     * {@code (resPos <= offset || resPos >= offset + intLen)}.
+     */
+    private final void primitiveRightShift(int n, int[] result, int resPos) {
```

Comment:
```suggestion
    private final void primitiveRightShift(int n, int[] result, int resOff) {
```
or something more similar to "offset".

Same for the left shift.

Reply:
What about `resFrom`, like the old `MBI.copyAndshift()` method?

Reply:
Up to you, the current `resPos` is OK as well.